Lib/glyphsLib/casting.py

```diff
@@ -53,312 +54,469 @@
+    def read(self, src):
+        return int(src, 16)
 
-    rx = '{%s}' % ', '.join(['([-.e\d]+)'] * dimension)
-    return [num(i) for i in re.match(rx, string).groups()]
+    def write(self, val):
+        assert isinstance(val, int)
+        return '%04X' % val
 
 
-def point(string):
-    return vector(string, 2)
+class RWVector(RWGlyphs):
+    """Read/write a vector in curly braces."""
 
+    def __init__(self, dimension):
+        self.dimension = dimension
+        self.regex = re.compile('{%s}' % ', '.join(['([-.e\d]+)'] * dimension))
 
-def transform(string):
-    return vector(string, 6)
+    def read(self, src):
+        """Parse a vector from a string with format {X, Y, Z, ...}."""
+        return [num.read(i) for i in self.regex.match(src).groups()]
 
+    def write(self, val):
+        assert isinstance(val, list) and len(val) == self.dimension
+        return '{%s}' % (', '.join(str(v) for v in val))
 
-def node(string):
-    """Cast a node from a string with format X Y TYPE [SMOOTH]."""
 
-    rx = '([-.e\d]+) ([-.e\d]+) (LINE|CURVE|OFFCURVE|n/a)(?: (SMOOTH))?'
-    m = re.match(rx, string).groups()
-    return [num(m[0]), num(m[1]), m[2].lower(), bool(m[3])]
+class RWPoint(RWVector):
+    """Read/write a two-element vector."""
+    def __init__(self):
+        RWVector.__init__(self, 2)
 
 
-def intlist(strlist):
-    return map(int, strlist)
+class RWTransform(RWVector):
+    """Read/write a six-element vector."""
 
+    def __init__(self):
+        RWVector.__init__(self, 6)
 
-def pointlist(strlist):
-    return map(point, strlist)
 
+class RWNode(RWGlyphs):
+    """Read/write a node on an outline."""
 
-def nodelist(strlist):
-    return map(node, strlist)
+    _regex = re.compile(
+        '([-.e\d]+) ([-.e\d]+) (LINE|CURVE|OFFCURVE|n/a)(?: (SMOOTH))?')
 
+    def read(self, src):
+        """Cast a node from a string with format X Y TYPE [SMOOTH]."""
+        g = self._regex.match(src).groups()
+        return [num.read(g[0]), num.read(g[1]), g[2].lower(), bool(g[3])]
 
-def glyphs_datetime(string):
-    """Parse a datetime object from a string."""
+    def write(self, val):
+        assert isinstance(val, list) and len(val) == 4
+        v2 = val[2]
+        # glyphs has this lower case
+        if v2 != 'n/a':
+            v2 = v2.upper()
+        return '%s %s %s%s' % (val[0], val[1], v2, ' SMOOTH' if val[3] else '')
 
-    # parse timezone ourselves, since %z is not always supported
-    # see: http://bugs.python.org/issue6641
-    string, tz = string.rsplit(' ', 1)
-    datetime_obj = datetime.datetime.strptime(string, '%Y-%m-%d %H:%M:%S')
-    offset = datetime.timedelta(hours=int(tz[:3]), minutes=int(tz[0] + tz[3:]))
-    return datetime_obj + offset
 
+class RWIntList(RWGlyphs):
+    """Read/write a list of ints."""
 
-def kerning(kerning_data):
-    """Cast the values in kerning data to ints."""
+    def read(self, src):
+        assert isinstance(src, list)
+        return map(int, src)
```

Comment:
In hindsight, we shouldn't use map like this because it doesn't work the same way in python3. So let's either do `list(map(...))` or (as I'd personally prefer) just use a list comprehension.